modules/wallet/siag.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+const (
+	// The header for all siag files. Do not change.
+	SiagFileHeader    = "siag"
+	SiagFileExtension = ".siakey"
+	SiagFileVersion   = "1.0"
+)
+
+var (
+	ErrInconsistentKeys     = errors.New("keyfiles provided that are for different addresses")
+	ErrInsufficientKeys     = errors.New("not enough keys provided to spend the siafunds")
+	ErrInsufficientSiafunds = errors.New("not enough siafunds in the keys provided to complete transction")
+	ErrNoKeyfile            = errors.New("no keyfile has been presented")
+	ErrUnknownHeader        = errors.New("file contains the wrong header")
+	ErrUnknownVersion       = errors.New("file has an unknown version number")
+)
+
+// A SiagKeyPair is the keypair used by siag.
+type SiagKeyPair struct {
+	Header           string
+	Version          string
+	Index            int // should be uint64 - too late now
+	SecretKey        crypto.SecretKey
+	UnlockConditions types.UnlockConditions
+}
+
+// SpendSiagSiafunds sends siafunds to another address. The siacoins stored in
+// the siafunds are sent to an address in the wallet.
+func (w *Wallet) SpendSiagSiafunds(amount types.Currency, dest types.UnlockHash, keyfiles []string) (txn types.Transaction, err error) {
+	if len(keyfiles) < 1 {
+		return types.Transaction{}, ErrNoKeyfile
+	}
+
+	// Load the siafund keys and verify they are sufficient to sign the
+	// transaction.
+	skps := make([]SiagKeyPair, len(keyfiles))
+	for i, keyfile := range keyfiles {
+		err = encoding.ReadFile(keyfile, &skps[i])
+		if err != nil {
+			return types.Transaction{}, err
+		}
+
+		if skps[i].Header != SiagFileHeader {
+			return types.Transaction{}, ErrUnknownHeader
+		}
+		if skps[i].Version != SiagFileVersion {
+			return types.Transaction{}, ErrUnknownVersion
+		}
+	}
+
+	// Check that all of the loaded files have the same address, and that there
+	// are enough to create the transaction.
+	baseUnlockHash := skps[0].UnlockConditions.UnlockHash()
+	for _, skp := range skps {
+		if skp.UnlockConditions.UnlockHash() != baseUnlockHash {
+			return types.Transaction{}, ErrInconsistentKeys
+		}
+	}
+	if uint64(len(skps)) < skps[0].UnlockConditions.SignaturesRequired {
+		return types.Transaction{}, ErrInsufficientKeys
+	}
+
+	// Check that there are enough siafunds in the key to complete the spend.
+	lockID := w.mu.RLock()
+	var availableSiafunds types.Currency
+	var sfoids []types.SiafundOutputID
+	for sfoid, sfo := range w.siafundOutputs {
+		if sfo.UnlockHash == baseUnlockHash {
+			availableSiafunds = availableSiafunds.Add(sfo.Value)
+			sfoids = append(sfoids, sfoid)
+		}
+		if availableSiafunds.Cmp(amount) >= 0 {
+			break
+		}
+	}
+	w.mu.RUnlock(lockID)
+	if availableSiafunds.Cmp(amount) < 0 {
+		return types.Transaction{}, ErrInsufficientSiafunds
+	}
+
+	// Truncate the keys to exactly the number needed.
+	skps = skps[:skps[0].UnlockConditions.SignaturesRequired]
```

Comment:
hmm. At this point I think we've both verified that there's at least 1 key and that there's also at least SignaturesRequired keys, but I'll look again.